src/main/java/org/skriptlang/skript/bukkit/potion/elements/conditions/CondHasPotion.java

```diff
@@ -0,0 +1,80 @@
+package org.skriptlang.skript.bukkit.potion.elements.conditions;
+
+import ch.njol.skript.conditions.base.PropertyCondition;
+import ch.njol.skript.conditions.base.PropertyCondition.PropertyType;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Example;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Condition;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.util.Kleenean;
+import org.bukkit.entity.LivingEntity;
+import org.bukkit.event.Event;
+import org.bukkit.potion.PotionEffect;
+import org.skriptlang.skript.bukkit.potion.util.SkriptPotionEffect;
+import org.skriptlang.skript.registration.SyntaxRegistry;
+import org.jetbrains.annotations.Nullable;
+
+@Name("Has Potion")
+@Description({
+	"Checks whether an entity has a potion effect with certain properties.",
+	"An entity is considered having a potion effect if it has a potion effect with at least the specified properties.",
+	"For example, if an entity has an 'ambient speed 5' effect, they would be considered as having 'speed 5'.",
+	"For exact comparisons, consider using the <a href='./expressions.html#ExprPotionEffect'>Potion Effect of Entity/Item</a>" +
+			" expression in an 'is' comparison."
+})
+@Example("""
+	if the player has a potion effect of speed:
+		message "You are sonic!"
+""")
+@Example("""
+	if all players have speed and haste active:
+		broadcast "This server is ready to mine!"
+""")
+@Since("2.6.1, INSERT VERSION (\"the\" support)")
+public class CondHasPotion extends Condition {
+
+	public static void register(SyntaxRegistry registry) {
+		PropertyCondition.register(registry, CondHasPotion.class, PropertyType.HAVE,
+				"%skriptpotioneffects% [active]",
+				"livingentities");
```

Comment:
CondContains conflicts? `%inventories% (has|have) %itemtypes%`

Reply:
CondContains has an earlier priority, so I think this should be okay? I can't think of a great syntax here, and I'd like to avoid something like `player has potion potion effect of speed`... type hints (once widely available) would resolve any issues like this. The optional `active` keyword exists for dealing with conflicts in the meantime - which is the approach I think we should prefer to take for these situations